Sources/SwiftOperatorPrecedence/PrecedenceGroup.swift

```diff
@@ -0,0 +1,120 @@
+//===------------------ PrecedenceGroup.swift -----------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+import SwiftSyntax
+
+/// Names a precedence group.
+///
+/// TODO: For now, we'll use strings, but we likely want to move this to
+/// a general notion of an Identifier.
+public typealias PrecedenceGroupName = String
```

Comment:
Just an idea: I’ve had good experience with declaring a new type
```swift
struct PrecedenceGroupName {
  let name: String
}
```

That way you can’t accidentally cast it to/from `String`, which would break if we switch the breaking storage later. 

That’s what I’m doing in `DiagnosticMessageID` at least.

Reply:
I rather hope we don't need a specific abstraction here in this library, but instead that we can introduce the notion of an `Identifier` into `SwiftSyntax` itself.

Reply:
OK, makes sense 👍